src/org/ods/services/OpenShiftService.groovy

```diff
@@ -611,6 +611,61 @@ class OpenShiftService {
         }
     }
 
+    /**
+     * Apply labels provided in <code>labels</code> to resources given in <code>resources</code> (format kind/name).
+     * The selection can be restricted to a project and also with a label selector given in <code>selector</code>.
+     * You can apply labels to all object kinds with <code>resources=all</code> and using a selector.
+     * If the label already exists, it will be overwritten.
+     * If the value for one label is the empty string, the label will be set with an empty string as value.
+     * If the value for one label is null, the label will be deleted from the selected resources.
+     *
+     * Allowed selector conditions are:
+     * <code>label=value</code>
+     * <code>label==value</code>
+     * <code>label!=value</code>
+     * More than one condition can be specified, separated by commas.
+     *
+     * @param project the project to apply the operation to. The current project, if null.
+     * @param resources the resources to apply the labels to, with format <code>kind[/name]</code>.
+     * @param labels a <code>Map</code> containing label keys and values.
+     * @param selector a comma-separated list of label conditions.
+     * @return the output of the shell script running the OpenShift client command.
+     * @throws IllegalArgumentException if no <code>resources</code> or no <code>labels</code> are provided.
+     */
+    String labelResources(String project, String resources, Map<String, String> labels, String selector = null) {
+        if (!resources) {
+            throw new IllegalArgumentException('You must specify the resources to label')
+        }
+        if (!labels) {
+            throw new IllegalArgumentException('At least one label update is required')
+        }
+        def labelStr = labels.collect { key, value ->
+            def label = key
+            label += value == null ? '-' : "='${value}'"
+            return label
+        }.join(' ')
+        if (logger.getDebugMode()) {
+            logger.debug("Setting labels ${labelStr} to resources ${resources} selected by ${selector}")
+        }
+        def script = "oc label --overwrite ${resources} "
```

Comment:
This might trigger a redeploy, right?

Reply:
In all tests I have done, labeling does not trigger redeploys. Otherwise, it is just a question of temporarily disabling automatic rollouts, but it doesn't seem to be needed at all.